llpc/tool/llpcCompilationUtils.cpp

```diff
@@ -274,14 +261,18 @@ Expected<std::string> assembleSpirv(const std::string &inFilename) {
   if (binSize < 0)
     return createResultError(Result::ErrorInvalidShader, Twine("Failed to assemble SPIR-V: \n") + log);
 
-  fwrite(spvBin.data(), 1, binSize, outFile);
+  // Caller is responsible for calling delete (note: this will normally happen
+  // as part of cleanupCompileInfo)
+  char *bin = new char[binSize]();
```

Comment:
This syntax looks a bit uncommon. Shouldn’t it just be `new char[binSize]` (no parentheses)?

Reply:
The parens make it value initialize array elements (to zero), similar to how you would force a local variable of primitive type to be initialized: https://en.cppreference.com/w/cpp/language/new#Construction

Reply:
I removed the parentheses - trying to get to bottom of asan failure. Of course, it was nothing to do with the issue, but I forgot to add them back in again (I'll push another change).